CMakeLists.txt

```diff
@@ -0,0 +1,258 @@
+
+set(gvk_AllowedLibraryTypes INTERFACE SHARED STATIC)
+set(gvk_LibraryType STATIC CACHE STRING
+        "The type of library gvk should be built as. Must be one of ${gvk_AllowedLibraryTypes}. Default: STATIC")
+set_property(CACHE gvk_LibraryType PROPERTY STRINGS ${gvk_AllowedLibraryTypes})
+
+if(NOT gvk_LibraryType IN_LIST gvk_AllowedLibraryTypes)
+    message(FATAL_ERROR "gvk_LibraryType must be one of ${gvk_AllowedLibraryTypes}")
+endif()
+
+set(gvk_IncludeScope INTERFACE)
+set(gvk_SourceScope INTERFACE)
+if (NOT gvk_LibraryType STREQUAL "INTERFACE")
+    set(gvk_IncludeScope PUBLIC)
+    set(gvk_SourceScope PRIVATE)
+endif (NOT gvk_LibraryType STREQUAL "INTERFACE")
+
+option(gvk_ForceAssimpBuild "Force and use a local build of Assimp even if it is installed on the system. (Linux only)" OFF)
+option(gvk_StaticDependencies "Build dependencies as static instead of shared libraries (Assimp & GLFW). (Linux only)" OFF)
+if (gvk_StaticDependencies)
+    set(BUILD_SHARED_LIBS OFF)
+endif()
+
+option(gvk_ReleaseDLLsOnly "Use release DLLs for all dependencies of examples. (Windows only)" ON)
+option(gvk_CreateDependencySymlinks "Create symbolic links instead of copying dependencies of examples, i.e. DLLs (Windows only) & assets." ON)
+
+option(gvk_BuildExamples "Build all examples for Gears-Vk." OFF)
+option(gvk_BuildHelloWorld "Build example: hello_world." OFF)
+option(gvk_BuildFramebuffer "Build example: framebuffer." OFF)
+option(gvk_BuildComputeImageProcessing "Build example: compute_image_processing." OFF)
+option(gvk_BuildMultiInvokeeRendering "Build example: multi_invokee_rendering." OFF)
+option(gvk_BuildModelLoader "Build example: model_loader." OFF)
+option(gvk_BuildOrcaLoader "Build example: orca_loader." OFF)
+option(gvk_BuildRayQueryInRayTracingShaders "Build example: ray_query_in_ray_tracing_shaders." OFF)
+option(gvk_BuildRayTracingWithShadowsAndAO "Build example: ray_tracing_with_shadows_and_ao." OFF)
+option(gvk_BuildRayTracingCustomIntersection "Build example: ray_tracing_custom_intersection." OFF)
+option(gvk_BuildTextureCubemap "Build example: texture_cubemap." OFF)
+option(gvk_BuildVertexBuffers "Build example: vertex_buffers." OFF)
+
+if (gvk_BuildExamples)
+    set(gvk_BuildHelloWorld ON)
+    set(gvk_BuildFramebuffer ON)
+    set(gvk_BuildComputeImageProcessing ON)
+    set(gvk_BuildMultiInvokeeRendering ON)
+    set(gvk_BuildModelLoader ON)
+    set(gvk_BuildOrcaLoader ON)
+    set(gvk_BuildRayQueryInRayTracingShaders ON)
+    set(gvk_BuildRayTracingWithShadowsAndAO ON)
+    set(gvk_BuildRayTracingCustomIntersection ON)
+    set(gvk_BuildTextureCubemap ON)
+    set(gvk_BuildVertexBuffers ON)
+endif()
+
+# ---------------------- Framework ------------------------
+
+add_library(${PROJECT_NAME} ${gvk_LibraryType})
+
+# Append MSVC specific compiler flags
+if (MSVC)
+    target_compile_options(${PROJECT_NAME}
+        ${gvk_SourceScope} "/bigobj"
+        ${gvk_SourceScope} "/MP")
+endif(MSVC)
```

Comment:
It looks like we need to set those compile options for _all_ projects, i.e., also the example applications. Building the library (`Gears_Vk`) works, but building `hello_world` fails with the following message:

``` 
I:\workwork\Gears-Vk\repo\auto_vk\src\avk.cpp : fatal error C1128: number of sections exceeded object file format limit: compile with /bigobj
  ninja: build stopped: subcommand failed.
```

Reply:
This should do the trick: 0f4b7ec